src/base-components/checkbox-base/index.js

```diff
@@ -0,0 +1,94 @@
+import React, { useEffect, useState, useRef } from 'react'
+import PropTypes from 'prop-types'
+
+import CheckboxGroupBase from './checkbox-group-base'
+
+
+const CheckboxBase = React.forwardRef(({
+  classes,
+  label,
+  checked,
+  defaultChecked,
+  indeterminate,
+  onChange,
+  inputProps,
+  isNestingGroup,
+}, ref) => {
+  const inputRef = useRef()
+  const [name, setName] = useState(null)
+  const [defaultCheck, setDefaultCheck] = useState(defaultChecked)
+  // TODO: debug inconsistency of using [defaultCheck] state for nested grouped checkboxes
+  const _defaultCheck = isNestingGroup ? defaultChecked : defaultCheck
+
+  useEffect(() => {
+    if (indeterminate !== null) {
+      inputRef.current.indeterminate = indeterminate
+    }
+  }, [indeterminate])
+
+  const handleChange = (e) => {
+    if (name === label || (defaultCheck && name === null)) {
+      setName('')
+    } else {
+      setName(e.target.name)
+    }
+    if (defaultCheck) {
+      setDefaultCheck(false)
+    }
+    if (onChange) {
+      onChange({ label: e.target.name, checked: e.target.checked })
+    }
+  }
+
+  return (
+    <div ref={ref} className={classes.root}>
+      <span className='self-center'><input
+        ref={inputRef}
+        className={`cursor-pointer ${classes.input}`}
+        type='checkbox'
+        name={label}
+        checked={_defaultCheck
+          ? _defaultCheck
+          : checked === null 
+            ? name === label
+            : checked}
+        onChange={handleChange}
+        {...inputProps}
+      /></span>
+      <span className='self-center align-middle'><label
+        className={classes.label}
+        htmlFor={label}
+      >{label}</label></span>
```

Comment:
what's the opinion on allowing labels to also be clickable and thus control the status of being checked?

the default HTML behavior with the following would allow clicking on the label to toggle the checked 

```html
<input type="checkbox" id="some-input" name="some-input" ... />
<label for="some-input"> text label</label>
```

the default virtual dom implementation of react/jsx also indicate so (with both `for` and `htmlFor`) -- quick sandbox https://codesandbox.io/s/nostalgic-rosalind-uhzy6?file=/src/App.js

however, the storybook doesn't seem the case. is it a design choice?

Reply:
No it should be clickable. In fact that's mapped in the prototype. My bad - maybe I should include a video next time.

Reply:
my bad left out `id`, added!

Reply:
tried with installing `uuid` but think its an overkill, so created a simple counter instead, lemme know if this is enough for this use case